crates/daemon/src/monitor.rs

```diff
@@ -478,10 +481,14 @@ impl Actor {
 
         tracing::trace!("Updating status of {num_transactions} transactions",);
 
-        let histories = self
-            .client
-            .batch_script_get_history(self.state.monitoring_scripts())
-            .context("Failed to get script histories")?;
+        let scripts = self.state.monitoring_scripts().cloned().collect();
+
+        let mut rx_script_history = batch_script_get_history(self.client.clone(), scripts);
+
+        let mut histories = Vec::new();
+        while let Some(script_history) = rx_script_history.recv().await {
+            histories.push(script_history)
+        }
 
```

Comment:
you could also add a poor man's check whether all the scripts got a result by comparing `anyhow::ensure!(scripts.len() == histories.len()` (or in a less dramatic way, by just logging instead of throwing an error.